### ISSUE TITLE
How can I get the strings (or more generally resources) referenced or used in an instruction?  

### ISSUE THREAD

**Issue by maelzawawy** [androguard/androguard#593]
How can I get the strings (or more generally resources) referenced or  used in an instruction?

**Comment by reox** [comment]
currently this is not really implemented but in other projects, See #591

**Comment by reox** [comment]
So this is actually somewhat implemented in androguard...

Use this example code to dump the AST of the APK:
```
from androguard.decompiler.dad.decompile import DvMachine
from pprint import pprint
from androguard.core.androconf import show_logging
import logging

show_logging(logging.WARNING)

default_file = 'examples/android/TestsAndroguard/bin/TestActivity.apk'
machine = DvMachine(default_file)

pprint(machine.get_ast())
```

The decompiled source code of TestSynthetic4 is:
```
public static void TestSynthetic4(int p3)
    {
        new tests.androguard.TestSynthetic$4(new Object(), p3).start();
        System.out.println("end");
        return;
    }
```

and the AST:
```
['BlockStatement', None,
	[['ExpressionStatement',
	  ['MethodInvocation',
	   [['ClassInstanceCreation',
		 [['ClassInstanceCreation',
		   [],
		   ['TypeName',
			('java/lang/Object',
			 0)]],
		  ['Local',
		   'p3']],
		 ['TypeName',
		  ('tests/androguard/TestSynthetic$4',
		   0)]]],
	   ('tests/androguard/TestSynthetic$4',
		'start',
		'()V'),
	   'start',
	   True]],
	 ['ExpressionStatement',
	  ['MethodInvocation',
	   [['FieldAccess',
		 [['TypeName',
		   ('java/lang/System',
			0)]],
		 ('java/lang/System',
		  'out',
		  'Ljava/io/PrintStream;')],
		['Literal',
		 'end',
		 ('java/lang/String',
		  0)]],
	   ('java/io/PrintStream',
		'println',
		'(Ljava/lang/String;)V'),
	   'println',
	   True]],
	 ['ReturnStatement', None]
	]
],
```

here the `System.out.println("end");` statement for better readability:
```
'ExpressionStatement', [
	'MethodInvocation', 
	[
		['FieldAccess', [['TypeName', ('java/lang/System', 0)]], ('java/lang/System', 'out', 'Ljava/io/PrintStream;')],
		['Literal', 'end', ('java/lang/String', 0)]
	],
	('java/io/PrintStream', 
	'println', 
	'(Ljava/lang/String;)V'), 
	'println', 
	True
]
```

So there is the `end` string in plain sight.
